implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -2,32 +2,43 @@ use super::map_multiaddr_err;
 use crate::error::ApiError;
 use crate::nodes::models::secure_channel::{
     CreateSecureChannelListenerRequest, CreateSecureChannelRequest, CreateSecureChannelResponse,
-    DeleteSecureChannelRequest, DeleteSecureChannelResponse, ShowSecureChannelRequest,
-    ShowSecureChannelResponse,
+    CredentialExchangeMode, DeleteSecureChannelRequest, DeleteSecureChannelResponse,
+    ShowSecureChannelRequest, ShowSecureChannelResponse,
 };
-use crate::nodes::registry::{IdentityRouteKey, SecureChannelInfo};
 use crate::nodes::NodeManager;
+use crate::DefaultAddress;
 use minicbor::Decoder;
 use ockam::identity::TrustEveryonePolicy;
 use ockam::{Address, Result, Route};
 use ockam_core::api::{Request, Response, ResponseBuilder};
-use ockam_core::AsyncTryClone;
+use ockam_core::{route, AsyncTryClone};
 use ockam_identity::{IdentityIdentifier, TrustMultiIdentifiersPolicy};
 use ockam_multiaddr::MultiAddr;
-use std::collections::HashSet;
-use std::sync::Arc;
 
 impl NodeManager {
-    pub(super) async fn create_secure_channel_impl<'a>(
+    async fn get_credential_if_needed(&self) -> Result<()> {
+        let identity = self.identity()?;
+
+        if identity.credential().await.is_some() {
+            return Ok(());
+        }
+
+        self.get_credential_impl(identity, false, "" /* FIXME */)
+            .await?;
```

Comment:
IIUC, calling this method without calling the `get_credential` method first will cause an error, right? So, somehow, when we have this `FIXME` figured out, we'll be able to skip that call from the command side, is that correct?

Reply:
```
So, somehow, when we have this FIXME figured out, we'll be able to skip that call from the command side, is that correct?
```
Correct

```
calling this method without calling the get_credential method first will cause an error, right?
```
Didn't get the question

Reply:
Both questions were more or less pointing in the same direction, so we can consider them both as answered :laughing: